src/fhda/tortoise.py

```diff
@@ -0,0 +1,256 @@
+import os
+import uuid
+import asyncio
+import copy
+from typing import List, Dict, Any, Optional, Callable
+import time
+import json
+from pydantic import BaseModel, Field
+from . import prompts
+
+from futurehouse_client import FutureHouseClient
+from futurehouse_client.models import TaskRequest, RuntimeConfig
+from futurehouse_client.models.app import AuthType
+
+
+class StepConfig(BaseModel):
+    """Configuration for a step in the pipeline"""
+
+    language: str = Field(
+        default="PYTHON", description="Language for execution environment"
+    )
+    max_steps: int = Field(
+        default=30, description="Maximum number of steps for the agent"
+    )
+    timeout: int = Field(default=15 * 60, description="Timeout for the step in seconds")
+    eval: bool = Field(default=True, description="Whether to use eval mode")
+
+
+class Step(BaseModel):
+    """A step in the pipeline"""
+
+    name: str = Field(
+        description="Name of the job to run (e.g. 'job-futurehouse-data-analysis-crow-high')"
+    )
+    prompt: str = Field(description="Prompt template to use for the step")
+    cot_prompt: bool = Field(
+        default=False, description="Whether to augment the query with COT prompting"
+    )
+    prompt_args: Dict[str, Any] = Field(
+        default_factory=dict, description="Arguments to format the prompt template"
```

Comment:
```suggestion
        default_factory=dict, description="Keyword arguments to format the prompt template."
```

Just being more explicit